numpy/distutils/unixccompiler.py

```diff
@@ -61,7 +61,7 @@ def UnixCCompiler__compile(self, obj, src, ext, cc_args, extra_postargs, pp_opts
         raise CompileError(msg)
 
     # add commandline flags to dependency file
-    with open(obj + '.d', 'a') as f:
+    with open(obj + '.d', 'w') as f:
```

Comment:
this has to append as it would truncate the dependency file otherwise
but it should not write when `deps` is empty and thus auto dependencies is off